### PR TITLE
Manual cherry pick of (#22119) to Release 7.1

### PR DESCRIPTION
**Pull request by pvev**
Manual cherry pick of  (#22119) to Release 7.1

* MM-49813 - validate members invitation permissions to private channels

* Validate on the receiver side that the sender id still has permissions to invite

* add translations

* fix texts

* simplify implementation

* remove local test hardcoded data

* add test to user_test

* fix vet warning

* Add comments to validate user permissions function

Co-authored-by: Martin Kraft <martin@upspin.org>

* regenerate app layer iface

* fix unit test

* fix app user_tests

---------

Co-authored-by: Martin Kraft <martin@upspin.org>

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/app_iface.go [hunk 1]
```diff
@@ -100,11 +100,13 @@ type AppIface interface {
 	// DefaultChannelNames returns the list of system-wide default channel names.
 	//
 	// By default the list will be (not necessarily in this order):
+	//
```
Comment (pvev) [inline app/app_iface.go:103]: these lines were added automatically as part of the `make app-layers`

File: app/app_iface.go [hunk 2]
```diff
@@ -377,9 +379,11 @@ type AppIface interface {
 	// UserIsInAdminRoleGroup returns true at least one of the user's groups are configured to set the members as
 	// admins in the given syncable.
 	UserIsInAdminRoleGroup(userID, syncableID string, syncableType model.GroupSyncableType) (bool, *model.AppError)
+	// ValidateUserPermissionsOnChannels filters channelIds based on whether userId is authorized to manage channel members. Unauthorized channels are removed from the returned list.
+	ValidateUserPermissionsOnChannels(userId string, channelIds []string) []string
 	// VerifyPlugin checks that the given signature corresponds to the given plugin and matches a trusted certificate.
 	VerifyPlugin(plugin, signature io.ReadSeeker) *model.AppError
-	//GetUserStatusesByIds used by apiV4
+	// GetUserStatusesByIds used by apiV4
```
Comment (pvev) [inline app/app_iface.go:386]: same here

**Comment by pvev** [comment]
not sure why this keep showing problems with the app layers, locally running the `make app-layers` command don't show any difference 🤔

**Comment by pvev** [comment]
@agnivade this PR is also giving some CI problems and as I mentioned in the previous comment, locally the app layers seems to be fine. Any idea what could be causing it to fail here?

**Comment by pvev** [comment]
> @agnivade this PR is also giving some CI problems and as I mentioned in the previous comment, locally the app layers seems to be fine. Any idea what could be causing it to fail here?

BTW, this is the cherry-pick for 7.1 that had to be done manually because of code variations from 7.1 to the latest version used for the solution implementation.

**Comment by agnivade** [comment]
Yeah .. all those old branches will fail due to an enterprise PR that is very difficult to cherry pick due to merge conflicts. Looks like 7.1 is an ESR, so it's a bit difficult. Unfortunately Ibrahim is on PTO so fixing the merge conflict might be tricky to fix right away. For now, I think we need to force merge. Sorry about this.

**Comment by pvev** [comment]
> Yeah .. all those old branches will fail due to an enterprise PR that is very difficult to cherry pick due to merge conflicts. Looks like 7.1 is an ESR, so it's a bit difficult. Unfortunately Ibrahim is on PTO so fixing the merge conflict might be tricky to fix right away. For now, I think we need to force merge. Sorry about this.

by Force merge, you mean locally doing the cherry-pick and pushing directly to the branch, or you mean just merge this one regardless of the CI process failing? @agnivade ?

**Comment by agnivade** [comment]
The latter. But only the ee/mattermost is allowed to fail. This PR needs to update the app layers as well.

**Comment by pvev** [comment]
> The latter. But only the ee/mattermost is allowed to fail. This PR needs to update the app layers as well.

thanks @agnivade , the weird thing with this one is that the app layers locally say they are up to date, but in the CI failing. Is there any other way to update the app-layer different than `make app-layers` ?

**Comment by agnivade** [comment]
I'll let you do some more debugging. Think about it, what can be different if both of us run `make app-layers` on the exact same codebase, but still get different output? For starters, you can print the difference in CI and see what it is that's different. Then you should be able to figure it out yourself.

We should actually print the difference in CI.

**Comment by pvev** [comment]
> I'll let you do some more debugging. Think about it, what can be different if both of us run `make app-layers` on the exact same codebase, but still get different output? For starters, you can print the difference in CI and see what it is that's different. Then you should be able to figure it out yourself.
> 
> We should actually print the difference in CI.

I can't make it print the difference, my gitlab account don't have access to the CI repos :(

**Comment by agnivade** [comment]
This is in CircleCI, not Gitlab. As part of the mattermost GitHub org, you should be able to access it.

**Comment by pvev** [comment]
thanks @agnivade on the guidance on how to find the app-layers error!

**Review by agnivade — APPROVED** [review]
Looks good sans the cherry-pick of https://github.com/mattermost/mattermost-server/pull/22174 to 7.1.

**Pull request merged**